rts/lra/lra-test/lra-test-arquillian-extension/src/main/java/io/narayana/lra/arquillian/spi/LRAAnnotationAdjuster.java

```diff
@@ -0,0 +1,206 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+
+package io.narayana.lra.arquillian.spi;
+
+import org.eclipse.microprofile.lra.annotation.ws.rs.LRA;
+import org.jboss.logging.Logger;
+
+import javax.ws.rs.core.Response;
+import java.lang.annotation.Annotation;
+import java.lang.reflect.Field;
+import java.lang.reflect.Method;
+import java.time.temporal.ChronoUnit;
+import java.util.Arrays;
+import java.util.Map;
+
+/**
+ * Helper class to do internal changes of annotations during runtime.
+ */
+public class LRAAnnotationAdjuster {
+    private static final Logger log = Logger.getLogger(LRAAnnotationAdjuster.class);
+    private static final String ANNOTATIONS_FIELD_NAME = "annotations";
+    private static final String ANNOTATION_DATA_METHOD_NAME = "annotationData";
+
+    /**
+     * Take the clazz, check if contains the {@link LRA} annotation.
+     * The LRA annotation is then replaced by wrapped {@link LRAWrapped}.
+     */
+    static void processWithClass(Class<?> clazz) {
+        LRA lraAnnotation = clazz.getDeclaredAnnotation(LRA.class);
+        if (lraAnnotation != null) {
+            LRAAnnotationAdjuster.adjustLRAAnnotation(clazz, lraAnnotation);
+        }
+        Arrays.stream(clazz.getMethods()).forEach(method -> {
+            LRA lraAnnotationMethod = method.getDeclaredAnnotation(LRA.class);
+            if (lraAnnotationMethod != null) {
+                LRAAnnotationAdjuster.adjustLRAAnnotation(method, lraAnnotationMethod);
+            }
+        });
+    }
+
+    /**
+     * Changing the LRA annotation declared on class by wrapping it with {@link LRAWrapped}.
+     */
+    static void adjustLRAAnnotation(Class clazzToLookFor, LRA originalLRAAnnotation) {
+        if (doesJDKDefineAnnotationsField()) {
+            // JDK7 has "annotations" field
+            try {
+                Field annotations = Class.class.getDeclaredField(ANNOTATIONS_FIELD_NAME);
+                annotations.setAccessible(true);
+                Map<Class<? extends Annotation>, Annotation> map =
+                        (Map<Class<? extends Annotation>, Annotation>) annotations.get(clazzToLookFor);
+                map.put(LRA.class, new LRAWrapped(originalLRAAnnotation));
+            } catch (Exception  e) {
+                e.printStackTrace();
+            }
+        } else {
+            try {
+                // JDK8+ has "annotationData" private method
+                // obtaining reference to private class AnnotationData
+                Method method = Class.class.getDeclaredMethod(ANNOTATION_DATA_METHOD_NAME, null);
```

Comment:
My IDE says this is a varargs param so you don't need to use null.

Reply:
+1